src/components/MapboxMap/index.tsx

```diff
@@ -52,12 +52,33 @@ const MapboxMap: React.FC<MapboxMapProps> = ({
 
   // Store map transform data in state so we can pass it to children
   const [transform, setTransform] = useState<MapboxMapTransform | null>(null);
+  // maintain this reference so we can access it in the useEffect below without including it in the dependency array
+  const transformRef = useRef(transform);
+  transformRef.current = transform;
 
   // Let the parent component overwrite the map bounds
   useDeepCompareEffectNoCheck(() => {
     if (!map || !fitBounds) return;
-    map.fitBounds(fitBounds.bounds, fitBounds.options);
-  }, [fitBounds]);
+    const { height, width } = map.getContainer().getBoundingClientRect();
```

Comment:
is this related to the renders?

Reply:
This is a separate bug I found while testing. I mentioned it in the description briefly.